### PR TITLE
Fix modeBarButtons issue

### PR DESCRIPTION
**Pull request by nickmelnikov82**
Fixes [dash](https://github.com/plotly/dash) issue [#1157](https://github.com/plotly/dash/issues/1157)
As @alexcjohnson wrote:
> The problem is [here](https://github.com/plotly/plotly.js/blob/b8905568ee672a60f1749a73dd8703835849087e/src/components/modebar/manage.js#L342) where we mutate the input - which is the user-provided array, something we should not alter. Instead we should create a copy with the object inserted in place of the string. Would be a one-line change if we used `ramda` in plotly.js, but we don't...

### CODE REVIEW

File: src/components/modebar/manage.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ var isUnifiedHover = require('../fx/helpers').isUnifiedHover;
 var createModeBar = require('./modebar');
 var modeBarButtons = require('./buttons');
 var DRAW_MODES = require('./constants').DRAW_MODES;
+var cloneDeep = require('lodash').cloneDeep;
```
Comment (archmoj) [inline src/components/modebar/manage.js:11]: Thanks very much for the PR.
Please note that [`lodash`](https://www.npmjs.com/package/lodash) is a dev-dependency. 
And for various reasons (e.g. size & potential security issues) we are not interested to include that in our bundles.
Alternatively you could use
```js
var extendDeep = require('../../lib').extendDeep;
```
and then 
```js
var customButtons = extendDeep({}, context.modeBarButtons);
```
Or perhaps even better would be to use `extendFlat` if possible?

Comment (nickmelnikov82) [reply src/components/modebar/manage.js]: Fixed. Using the extendFlat function is not possible because the object has a nested structure.

**Comment by archmoj** [comment]
Please add a test here: https://github.com/plotly/plotly.js/blob/042742d67638ea970bd70d5c143db3a181e613bc/test/jasmine/tests/modebar_test.js#L158 

Also I appreciate if you draft `6177_fix.md` as described here:
https://github.com/plotly/plotly.js/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Thank you!

File: src/components/modebar/manage.js [hunk 2]
```diff
@@ -44,7 +45,7 @@ module.exports = function manageModeBar(gd) {
         ].join(' '));
     }
 
-    var customButtons = context.modeBarButtons;
+    var customButtons = extendDeep([], context.modeBarButtons);
```
Comment (archmoj) [inline src/components/modebar/manage.js:48]: This deep copy appears to be needed in the first case only.
I suggest you revert the change here and move deep copy inside `fillCustomButton` function.
For example:
```js
function fillCustomButton(inButtons) {
  var customButtons = extendDeep([], inButtons);
  ...
```

**Comment by archmoj** [comment]
Thanks very much for the PR.
:dancer:
